src/render/processor.rs

```diff
@@ -61,7 +61,14 @@ impl<'a> AudioParamValues<'a> {
 
     /// Get the computed values for the given [`crate::param::AudioParam`]
     ///
-    /// For both A & K-rate params, it will provide a slice of length [`crate::RENDER_QUANTUM_SIZE`]
+    /// # Warning
+    ///
+    /// In the current implementation, the slice will always have a length of
+    /// [`RENDER_QUANTUM_SIZE`](crate::RENDER_QUANTUM_SIZE),
+    /// but one should not rely on this. Following the AudioWorklet specification
+    /// (cf. <https://www.w3.org/TR/webaudio/#audioworkletprocess-callback-parameters>),
+    /// a slice of length 1 may be returned in the future if all values in the
+    /// current quantum are identical (e.g. for k-rate processing).
     pub fn get(&self, index: &AudioParamId) -> &[f32] {
```

Comment:
I think in our previous discussion on this signature, we concluded that is very convenient for the end user to always return a slice of size 128.
We could consider to drop the warning and commit to it. It would require a breaking change though then, to change it back but that is a risk I'm willing to make. For now I don't see much drawback in resizing the array to 128 always, this should translate to a memcpy which should be very efficient

Reply:
yup you are right, reverted back to the previous doc